docs/src/Button.doc.js

```diff
@@ -12,217 +12,390 @@ const card = c => cards.push(c);
+        type: `'gray' | 'red' | 'blue' | 'transparent' | 'white'`,
+        required: false,
         defaultValue: 'gray',
-        href: 'color',
-      },
-      {
-        name: 'disabled',
-        type: 'boolean',
-        defaultValue: false,
+        description: 'Primary colors to apply to the button background.',
         href: 'combinations',
       },
       {
-        name: 'selected',
+        name: 'disabled',
         type: 'boolean',
-        defaultValue: false,
+        required: false,
+        defaultValue: null,
+        description:
+          'Set disabled state so buttons look inactive, cannot be interacted with, and actions are not available.',
         href: 'combinations',
       },
       {
         name: 'iconEnd',
         type: '$Keys<typeof icons>',
-        description:
-          'Add an icon to be displayed after the button text. This allows type checking for a valid icon name based on the keys from the list of icons in Icon.',
+        required: false,
+        defaultValue: false,
+        description: [
+          'Add a Gestalt icon to be displayed after the button text. Sometimes an icon can help clarify the usage of a button. Menus are a common use case.',
+          'Accessibility: Icons on buttons are not accessible for screen readers.',
+          'Use IconButton if you only need buttons with icons and no text.',
+        ],
         href: 'iconEnd',
       },
       {
         name: 'inline',
         type: 'boolean',
+        required: false,
         defaultValue: false,
-        href: 'widths',
-      },
-      {
-        name: 'name',
-        type: 'string',
+        description:
+          'Display a button as an inline element so that it does not start on a new line breaking the flow of the content. Inline buttons are sized by the text within the button, whereas the default block buttons expand to the full width of their container.',
+        href: 'width',
       },
       {
         name: 'onClick',
         type: '({ event: SyntheticMouseEvent<> }) => void',
-      },
-      {
-        name: 'ref',
-        type: "React.Ref<'button'>",
-        description: 'Forward the ref to the underlying button element',
-        href: 'refExample',
+        required: false,
+        defaultValue: null,
+        description: [
+          'Callback fired when a button component is clicked (pressed and released) with a mouse or keyboard.',
+          'Required with button-role + button-type buttons.',
+        ],
+        href: 'selected',
       },
       {
         name: 'size',
-        type: `"sm" | "md" | "lg"`,
+        type: `'sm' | 'md' | 'lg'`,
+        required: false,
         defaultValue: 'md',
-        description: 'sm: 32px, md: 40px, lg: 48px',
-        href: 'combinations',
+        description:
+          'Display a button in different sizes. Size only changes top & bottom padding modifying the component height: sm (32px), md (40px), lg (48px).',
+        href: 'height',
```

Comment:
@AlbertCarreras could we rename `height` to `sizes`. It not only updates the top/bottom padding but also left/right.

Reply:
DONE Yeah, didn't notice lg also changes left/right padding